lib/beanie/registry_api/registry.ex

```diff
@@ -27,11 +27,17 @@ defmodule Beanie.RegistryAPI.Registry do
     url = url(registry, path)
     Logger.debug("Fetching registry url #{url}")
 
-    response = registry.http_client.get!(
-      url(registry, path),
-      [basic_auth: {registry.user, registry.password}]
-    )
-    Logger.debug("Got '#{response.body}' from #{url}")
-    Poison.decode!(response.body)
+    try do
+      response = registry.http_client.get!(
+        url(registry, path),
+        [basic_auth: {registry.user, registry.password}]
+      )
+      Logger.debug("Got '#{response.body}' from #{url}")
+      Poison.decode!(response.body)
+    rescue
+      e in HTTPotion.HTTPError -> e
+      %{"repositories" => [], :error => e}
```

Comment:
As said above, should b a way to do this without rescue, but FYI this should be like

```
rescue
   e in HTTPotion.HTTPError -> %{"repositories" => [], :error => e}
end
```